### PR TITLE
Update kubernetes manifests to ensure piped container runs as a non root user

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Part of https://github.com/pipe-cd/pipe/issues/987

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: manifests/piped/templates/deployment.yaml [hunk 1]
```diff
@@ -59,6 +61,12 @@ spec:
         - name: piped-config
           configMap:
             name: {{ include "piped.configMapName" . }}
+        - name: ssh-config
```
Comment (nghialv) [inline manifests/piped/templates/deployment.yaml:64]: Do we need this volume?

Comment (nghialv) [reply manifests/piped/templates/deployment.yaml:64]: Btw, we should change the default value of `ssh config path`.
https://github.com/pipe-cd/pipe/blob/master/pkg/config/piped.go#L165

By switching to a non-root user, piped will be failed with the current default value.

**Comment by nakabonne** [comment]
Fixed

**Pull request merged**